kapitan/targets.py

```diff
@@ -306,6 +306,43 @@ def load_target_inventory(inventory_path, targets):
     return target_objs
 
 
+def search_targets(inventory_path, targets, labels):
+    """returns a list of targets where the labels match, otherwise just return the original targets"""
+    if not labels:
+        return targets
+
+    try:
+        labels_dict = dict(label.split('=') for label in labels)
+    except ValueError:
+        logger.error("Compile error: Failed to parse labels, should be formatted like: kapitan compile -l env=prod app=example")
+        sys.exit(1)
+
+    targets_found = []
+    inv = inventory_reclass(inventory_path)
+
+    for target_name in inv['nodes']:
+        matched_all_labels = False
+        for label, value in labels_dict.items():
+            try:
+                if inv['nodes'][target_name]['parameters']['kapitan']['labels'][label] == value:
+                    matched_all_labels = True
+                    continue
+            except KeyError:
+                logger.debug("search_targets: label %s=%s didn't match target %s", label, value, target_name)
+
+            matched_all_labels = False
+            break
+
+        if matched_all_labels:
+            targets_found.append(target_name)
+
+    if len(targets_found) == 0:
+        logger.error("No targets found with labels: {}".format(labels))
```

Comment:
raise CompileError

Reply:
Same here, no stacktrace required as it's just a message, similar to what kubectl returns if no objects are found with labels.